flox/xrdtypes.py

```diff
@@ -38,7 +38,7 @@ def __eq__(self, other):
 PROMOTE_TO_OBJECT = [
     {np.number, np.character},  # numpy promotes to character
     {np.bool_, np.character},  # numpy promotes to character
-    {np.bytes_, np.unicode_},  # numpy promotes to unicode
+    {np.bytes_, np.str_},  # numpy promotes to unicode
```

Comment:
I don't think this is used? If not, we can delete...

Reply:
indeed, it is not: the only mention of `PROMOTE_TO_OBJECT` is its definition (in `xarray` it would have been used by `result_type`, but you didn't vendor that)